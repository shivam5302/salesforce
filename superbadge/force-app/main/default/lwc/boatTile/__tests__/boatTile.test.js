import { createElement } from 'lwc';
import BoatTile from 'c/boatTile';

const flushPromises = async () => new Promise((resolve) => setTimeout(resolve, 0));

describe('c-boat-tile', () => {
	let element;
	beforeEach(() => {
		element = createElement('c-boat-tile', {
			is: BoatTile
		});
		element.boat = {
			Id: '123',
			Name: 'Boaty McBoatFace',
			Description__c: 'This boat is the best!',
			Picture__c: 'http://www.example.com/boat.jpg',
			Length__c: 20,
			Contact__r: { Name: 'Contact Name' },
			BoatType__r: { Name: 'type' },
			Price__c: 300,
			Year_Built__c: 2019
		}
		document.body.appendChild(element);
	});

	afterEach(() => {
		// The jsdom instance is shared across test cases in a single file so reset the DOM
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
	});

	it('TODO: test case generated by CLI command, please fill in test logic', () => {

		// Assert
		// const div = element.shadowRoot.querySelector('div');
		expect(element).toMatchSnapshot();
	});

	it('Should dispatch event onClick card', async () => {
		//Given
		const boatselectHandler = jest.fn();
		element.addEventListener('boatselect', boatselectHandler);
		await flushPromises();
		const div = element.shadowRoot.querySelector("div");


		//When
		div.click();

		//Then
		await flushPromises();
		expect(boatselectHandler).toHaveBeenCalled();
		const eventDetail = boatselectHandler.mock.calls[0][0].detail;
		expect(eventDetail.boatId).toBe('123');
		expect(element).toMatchSnapshot()

	});

});