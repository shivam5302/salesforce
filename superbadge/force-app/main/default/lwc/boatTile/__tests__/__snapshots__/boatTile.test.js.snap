// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-boat-tile Should dispatch event onClick card 1`] = `
<c-boat-tile
  __lwc_scope_token__=""
>
  #shadow-root(open)
    <div
      __lwc_scope_token__=""
      class="tile-wrapper selected"
    >
      <div
        __lwc_scope_token__=""
        class="tile"
        style="background-image: url(http://www.example.com/boat.jpg);"
      />
      <div
        __lwc_scope_token__=""
        class="lower-third"
      >
        <h1
          __lwc_scope_token__=""
          class="slds-truncate slds-text-heading_medium"
        >
          Boaty McBoatFace
        </h1>
        <h2
          __lwc_scope_token__=""
          class="slds-truncate slds-text-heading_small"
        >
          Contact Name
        </h2>
        <lightning-formatted-number
          __lwc_scope_token__=""
        >
          #shadow-root(open)
        </lightning-formatted-number>
        <div
          __lwc_scope_token__=""
          class="slds-text-body_small"
        >
          20
        </div>
        <div
          __lwc_scope_token__=""
          class="slds-text-body_small"
        >
           type
        </div>
      </div>
    </div>
</c-boat-tile>
`;

exports[`c-boat-tile TODO: test case generated by CLI command, please fill in test logic 1`] = `
<c-boat-tile
  __lwc_scope_token__=""
>
  #shadow-root(open)
    <div
      __lwc_scope_token__=""
      class="tile-wrapper"
    >
      <div
        __lwc_scope_token__=""
        class="tile"
        style="background-image: url(http://www.example.com/boat.jpg);"
      />
      <div
        __lwc_scope_token__=""
        class="lower-third"
      >
        <h1
          __lwc_scope_token__=""
          class="slds-truncate slds-text-heading_medium"
        >
          Boaty McBoatFace
        </h1>
        <h2
          __lwc_scope_token__=""
          class="slds-truncate slds-text-heading_small"
        >
          Contact Name
        </h2>
        <lightning-formatted-number
          __lwc_scope_token__=""
        >
          #shadow-root(open)
        </lightning-formatted-number>
        <div
          __lwc_scope_token__=""
          class="slds-text-body_small"
        >
          20
        </div>
        <div
          __lwc_scope_token__=""
          class="slds-text-body_small"
        >
           type
        </div>
      </div>
    </div>
</c-boat-tile>
`;
